### ISSUE TITLE
There are multiple places to enter configuration

### ISSUE THREAD

**Issue by craig8** [GridOPTICS/GOSS#9]

**Comment by craig8** [comment]
I think that this will be true because of the way osgi works, everything is configurable through the file.  However I think that there should be a way to change things so that we can use placeholders easier.

    my.property = ${abc}

There should be a production environment file **goss.environment.properties** that is read on startup of felix and substitutes.

There should be a development environment file **goss.dev.properties** that is for development.

I think there should be a **default.environment.properties** file that is checked in to the repository that can be copied to **goss.environment.properties** and be used as a starting point.

To recap

 - Each bundle will still have its own property file capability (whether the bundle chooses to be part of the configuration is up to the bundle).
 - There is a global search and replace that happens during runtime to replace the ${..} values with values that are specified in goss.environtment.properties and goss.dev.properties.

**Comment by craigpnnl** [comment]
Closing as already addressed, via a different mechanism than the 2015 proposal.

The `${...}` placeholder substitution this issue asked for works today: `pnnl.goss.core.runner/goss-core.bndrun`'s `-runproperties` block injects values (activemq.host, openwire.port, felix.fileinstall.dir=conf) at OSGi framework start, and Felix FileInstall feeds the `conf/*.cfg` files through ConfigurationAdmin, which resolves the placeholders (for example `goss.openwire.uri = tcp://${activemq.host}:${openwire.port}`). A defensive `getProperty(key, default)` pattern falls back to a default when a value is still unsubstituted.

The specific proposed files (goss.environment.properties, goss.dev.properties, default.environment.properties) were never added, but the intent (one place to substitute runtime values) is satisfied for the OSGi path.

One residual, narrower concern noted for potential follow-up: the non-OSGi standalone path (GossSimpleRunner) has its own separate System-property/env-var/file-search config mechanism, so config-surface duplication persists in a new form. That is a distinct, smaller concern from the original placeholder request and can be tracked separately if it matters.
